courses/api.py

```diff
@@ -1339,18 +1339,48 @@ def import_courserun_from_edx(  # noqa: C901, PLR0913
     "program": "Program",
 }
 
+# Maps the value of settings.ENVIRONMENT to the hostname for that environment's Learn instance
+# This is ugly, if anyone has other suggestions I'm all ears.
+ENV_TO_LEARN_HOSTNAME_MAP = {
+    "production": "learn.mit.edu",
+    "rc": "learn.rc.mit.edu",
+    "ci": "learn.ci.mit.edu",
+}
+
 
 def get_verifiable_credentials_payload(certificate: BaseCertificate) -> dict:
-    # TODO: Need to figure out how to construct URLs correctly. They're used as a required ID.
+    # TODO: We could optimize these queries #noqa: TD002, TD003, FIX002
+    # It's not a massive priority though, as we have a total of 20k certs in prod as of 12/25
```

Comment:
watch out: @rachellougee will be adding many more legacy certs in January, maybe ~300K?

Reply:
Gotcha, thanks for the info. Think we might be in the clear then - based on a quick read, it looks like the certificates create path uses `bulk_create`. Bulk_create [doesn't trigger signals](https://docs.djangoproject.com/en/6.0/ref/models/querysets/#bulk-create), we won't see the program certificate path that provisions VCs fire, and we only provision VCs for CourseRunCertificates in the celery task.